huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop_name_rule.go

```diff
@@ -0,0 +1,238 @@
+		UpdateContext: resourceDesktopNameRuleUpdate,
+		DeleteContext: resourceDesktopNameRuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name_prefix": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"digit_number": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"start_number": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"single_domain_user_increment": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"is_default_policy": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+			"is_contain_user": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceDesktopNameRuleCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v2/{project_id}/desktop-name-policies"
+	)
+
+	client, err := cfg.NewServiceClient("workspace", region)
+	if err != nil {
+		return diag.Errorf("error creating Workspace client: %s", err)
+	}
+
+	createPath := client.Endpoint + httpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", client.ProjectID)
+	createOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         utils.RemoveNil(buildDesktopNameRuleBodyParams(d)),
+	}
+	requestResp, err := client.Request("POST", createPath, &createOpt)
+	if err != nil {
+		return diag.Errorf("error creating desktop name rule: %s", err)
+	}
+
+	respBody, err := utils.FlattenResponse(requestResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	policyId := utils.PathSearch("policy_id", respBody, "")
+	d.SetId(policyId.(string))
+
+	return resourceDesktopNameRuleRead(ctx, d, meta)
+}
+
+func buildDesktopNameRuleBodyParams(d *schema.ResourceData) map[string]interface{} {
+	return map[string]interface{}{
+		"policy_name":            d.Get("name"),
+		"name_prefix":            d.Get("name_prefix"),
+		"digit_number":           d.Get("digit_number"),
+		"start_number":           d.Get("start_number"),
+		"single_domain_user_inc": d.Get("single_domain_user_increment"),
+		"is_default_policy":      d.Get("is_default_policy"),
+	}
+}
+
+func resourceDesktopNameRuleRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg    = meta.(*config.Config)
+		region = cfg.GetRegion(d)
+	)
+
+	client, err := cfg.NewServiceClient("workspace", region)
+	if err != nil {
+		return diag.Errorf("error creating Workspace client: %s", err)
+	}
+
+	rule, err := GetDesktopNameRule(client, d.Id())
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "Workspace name rule")
```

Comment:
```suggestion
		return common.CheckDeletedDiag(d, err, "desktop name rule")
```
desktop or workspace? pls check the other places

Reply:
fixed